### ISSUE TITLE
TreeTable - Accessibility issue with role of table

### ISSUE THREAD

**Issue by antsteyer** [primefaces/primevue#7708]
### Describe the bug

TreeTable component is not accessible due the role `table` not compatible with the behavior of a treegrid element and the expanded rows. The role should be `treegrid` in order to use the attribute `aria-expanded` on the rows.

See https://dequeuniversity.com/rules/axe/4.10/aria-conditional-attr for reference.

![Image](https://github.com/user-attachments/assets/12436cc8-058a-45c8-9888-bc558453a35e)

Accessibility is very important for our business, please consider merging the PR quick if possible, thanks !

### Pull Request Link

https://github.com/primefaces/primevue/pull/7707

### Reason for not contributing a PR

- [ ] Lack of time
- [ ] Unsure how to implement the fix/feature
- [ ] Difficulty understanding the codebase
- [ ] Other

### Reproducer

https://primevue.org/treetable/

### Environment

Chrome with lighthouse

### Vue version

3.5.13

### PrimeVue version

4.3.4

### Steps to reproduce the behavior

1. Go to https://primevue.org/treetable/
2. Open developer console
3. Run Lighthouse - Accessibility

### Expected behavior

TreeTable component should be accessible
